utils/system_locale/src/linux.rs

```diff
@@ -0,0 +1,79 @@
+#[cfg(target_os = "linux")]
+mod linux_locale {
+    use locale::{
+        self,
+        linux::ffi::{
+            LC_ADDRESS, LC_ALL, LC_COLLATE, LC_CTYPE, LC_IDENTIFICATION, LC_MEASUREMENT,
+            LC_MESSAGES, LC_MONETARY, LC_NAME, LC_NUMERIC, LC_PAPER, LC_TELEPHONE, LC_TIME,
+        },
+    };
+    use std::{
+        collections::{HashMap, HashSet},
+        ffi::CStr,
+        ptr::null,
+    };
+
+    pub unsafe fn fetch_locale_settings() -> HashMap<i32, String> {
+        let locale_ptr = locale::linux::ffi::setlocale(LC_ALL, null());
+        let c_str: &CStr = CStr::from_ptr(locale_ptr as *const u8);
+        let mut locale_map = HashMap::new();
+
+        if let Ok(str_slice) = c_str.to_str() {
+            if str_slice.contains(';') {
```

Comment:
`contains` requires a full scan of the string, try to write the algorithm with just `split`.